engine/src/board.rs

```diff
@@ -343,7 +358,14 @@ impl Board {
     }
 
     pub fn spawns_left(&self, color: &Color, game_type: GameType) -> bool {
-        if self.reserve(color, game_type).iter().filter(|(_, v)| **v > 0).collect::<HashMap<&Bug, &i8>>().is_empty() || self.spawnable_positions(color).is_empty() {
+        if self
+            .reserve(color, game_type)
+            .iter()
+            .filter(|(_, v)| **v > 0)
+            .collect::<HashMap<&Bug, &i8>>()
+            .is_empty()
+            || self.spawnable_positions(color).is_empty()
```

Comment:
for massive boolean statements like this, i like to create descriptive variables for each clause. like:
```suggestion
        let has_reserve_bugs = self.reserve(color, game_type)
            .iter()
            .filter(|(_, v)| **v > 0)
            .count() > 0;
        let has_spawnable_positions = self.spawnable_positions(color).len() > 0;
        return has_reserve_bugs && has_spawnable_positions;
```